### ISSUE TITLE
Changing the visibility of a module parameter breaks `with`

### ISSUE THREAD

**Issue by brprice** [agda/agda#2668]
I would expect that within a module `M`, the following are more-or-less equivalent:
```agda
module M (A : Set) where ...
```
```agda
module M {A : Set} where ...
```
```agda
module M where
  postulate A : Set
  ...
```

However, a combination of module renaming and `with` shows they are not (postulate and hidden works, visible does not):
```agda
module Bug where

data _==_ {l}{X : Set l}(x : X) : X -> Set l where
  refl : x == x
{-# BUILTIN EQUALITY _==_ #-}
{-# BUILTIN REFL refl #-}

data Nat : Set where
  ze : Nat
  su : Nat -> Nat

-- has to pattern match: `badRefl _ = refl` typechecks
badRefl : (n : Nat) -> n == n
badRefl ze = refl
badRefl (su n) = refl

module Bug (Q : Set) where
  -- has to wrap: `Wrap = Nat` (and changing uses of wrap) typechecks
  data Wrap : Set where
    wrap  : Nat -> Wrap

  record Rec (A : Set) : Set where
    field
      recW : Nat -> Wrap

    foo : A -> Wrap -> Wrap
    foo _ (wrap i) = recW i

  THIN : Rec Nat
  THIN = record { recW = wrap }

  module Th = Rec THIN

  huh : forall (th : Nat)(t : Wrap)
-- If we don't go via Th, it typechecks
--   -> Rec.foo THIN th t == Rec.foo THIN th t
     -> Th.foo th t == Th.foo th t
  huh th t with (badRefl th)
  ... | p = refl
```

Agda (both 2.5.2 and 2.6.0-837f5d9) complains that
```
Checking Bug (/tmp/bug/Bug.agda).
/tmp/bug/Bug.agda:39,3-17
Expected a hidden argument, but found a visible argument
when checking that the type
(th : Nat) →
th == th →
(x : Set) (t : Wrap) →
Rec.foo (record { recW = wrap }) th t ==
Rec.foo (record { recW = wrap }) th t
of the generated with function is well-formed
```

Note that the parameter Q is completely unused (and removing it altogether makes agda accept the above).
This seems a bit tricky to tickle: changing any one of the three things above makes agda happy.
I originally noticed this in using `rewrite`, I guess the same reason `with` fails makes `huh th t rewrite (badRefl th) = refl` fail also.

**Comment by brprice** [comment]
This seems to have been fixed in 2.5.3 (specifically, commit 8518b8e seems to have solved this, along with #2727 and #2726)

**Comment by andreasabel** [comment]
Thanks for noticing!  I close...
